lib/graphql/define/assign_object_field.rb

```diff
@@ -18,16 +18,18 @@ def self.call(owner_type, name, type_or_field = nil, desc = nil, function: nil,
         base_field = if type_or_field.is_a?(GraphQL::Field)
           type_or_field.redefine(name: name_s)
         elsif function
-          GraphQL::Field.define(
+          kwargs = {
             arguments: function.arguments,
-            complexity: function.complexity,
             name: name_s,
             type: function.type,
             resolve: function,
             description: function.description,
             function: function,
             deprecation_reason: function.deprecation_reason,
-          )
+          }
+          kwargs[:complexity] = function.complexity if function.complexity
```

Comment:
Complexity can't be null here if youre passingr a function instance.

The reason its null is because you're passing a Function class and not an instance.

I think we should probably fail hard in that case since it would probably end up failing elsewhere if we accept Function classes here?

Reply:
Ah! That makes sense, I totally overlooked that issue! I guess it would make sense to fail if you're not passing an instance.